### ISSUE TITLE
When using TFLite for network inference on mobile phone, the shallow layer of same model takes more time to infer than the deep layer.

### ISSUE THREAD

**Issue by TXQGIT** [tensorflow/tensorflow#23213]
**System information**
- Have I written custom code:
N/A
- OS Platform and Distribution:
Ubuntu14.04, Win 10, Android Studio 3.0.1, sdk version 26
- Mobile device:
Gome K1 (CPU: MTK MT6757,  2.3GHZ)
- TensorFlow installed from (source or binary):
binary
- TensorFlow version (use command below):
1.11.0
- Python version:
2.7
- Bazel version (if compiling from source):
N/A
- GCC/Compiler version (if compiling from source):
N/A
- CUDA/cuDNN version:
N/A
- GPU model and memory:
N/A

**Describe the current behavior**
I am using the [offical demo(Java interface)](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/lite/java/demo) of TFLite on Android, which classify the frame captured by the camera.  The model is offical released [Mobilenet_v1_1.0_224 ](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/g3doc/models.md). The inference time is about 240ms in my mobile phone. Now, I need to get the feature maps of the last convolution layer, so I get the new tflite interpreter based on [TF Lite converter ](https://www.tensorflow.org/lite/convert/cmdline_examples) and the frozen graphs in the offical released  [Mobilenet_v1_1.0_224 ](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/g3doc/models.md), the comandline that I used is:
`tflite_convert --graph_def_file=mobilenet_v1_1.0_224_frozen.pb --output_file=mobilenet_v1_1.0_224.tflite --input_shapes=1,224,224,3 --input_array=input --output_arrays=MobilenetV1/MobilenetV1/Conv2d_13_pointwise/BatchNorm/FusedBatchNorm`
When I use the new tflite interpreter for inference to get the feature maps of the last conv layer, **the inference time is 320ms, which is more than the original 240ms when get the prob.** This makes me very confused, since the global avg pooling and FC is no longer needed when using the new interpreter to get the feature map of last conv layer and the compution costs is decreased, so the inference time should not increase. By the way, the code for time measurement remains unchanged, which is:
```
// Here's where the magic happens!!!
long startTime = SystemClock.uptimeMillis();
runInference();
long endTime = SystemClock.uptimeMillis();
Log.d(TAG, "Timecost to run model inference: " + Long.toString(endTime - startTime));
```
I also test the inference time when I get the feature maps of the middle conv layers: 
--------layer--------------------shape of feature map------------inference time(ms)
-------prob------------------------1x1001-------------------------------~240
conv2d_13_pointwise-----------7x7x1024------------------------------~320
conv2d_10_pointwise-----------14x14x512----------------------------~340
conv2d_6_pointwise------------14x14x512----------------------------~260
conv2d_5_pointwise------------28x28x256----------------------------~380
conv2d_3_pointwise------------56x56x128----------------------------~600

**It seems that the larger the feature map is, the bigger the inference time is,** and the effect of depth on inferenc time is much smaller. 
So, What is the reason? Any help would be grateful. Thanks!

**Exact command to reproduce:**
N/A

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Exact command to reproduce

**Comment by TXQGIT** [comment]
hanks, The problem description has been updated according to the issue template.

**Comment by brucechou1983** [comment]
When running a semantic segmentation model on an android device, I run into another issue that `readMultiDimentionalArray()` in `copyTo()` costs a lot of time.

When using android profiler, it looks like this:
![image](https://user-images.githubusercontent.com/8468820/48178309-db8c5d00-e353-11e8-8b8d-df4d33f6ebd5.png)

In the figure, `copyTo()` costs totally 88.22ms. `shapeOf()` costs 7.14ms and `readMultiDimentionalArray()` costs 81.07ms. The shape of the output tensor is (1, 128, 128, 1). 

Is there possible solution to reduce the time of this function?

**Comment by andrehentz** [comment]
multi-dimensional arrays incur an expensive copy in the interface between java and C++. Refer to 
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/java/src/main/java/org/tensorflow/lite/Interpreter.java#L217

>  ByteBuffer is the preferred way to pass large input data

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by suharshs** [comment]
Closing due to inactivity.
